### PR TITLE
feat: add stripe checkout redirect

### PR DESCRIPTION
**Pull request by ahmaxed**

### CODE REVIEW

**Review by naomi-lgbt — DISMISSED** [review]
LGTM~! 🥳 Was able to set up Ghost locally and test the Stripe redirect, which appears to work as intended. Could not figure out how to set up the internationalisation though... (I'm sure I'm missing something)

Thank you for your pull request! We look forward to additional contributions!

File: partials/banner.hbs [hunk 1]
```diff
@@ -1,5 +1,86 @@
...
     <p>
-        {{{t "Learn to code — <span>free 3,000-hour curriculum</span>"}}}
+        {{{t "Help our nonprofit pay for servers. <span>Donate Now</span>."}}}
     </p>
 </a>
+<script>
+    // Conditionally load stripe
+    const loadStripeList = ['en'];
+    console.log(loadStripeList.includes(`{{@site.lang}}`));
+    console.log(`{{@site.lang}}`);
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
```

File: partials/banner.hbs [hunk 2]
```diff
@@ -1,5 +1,86 @@
...
+        const donatePageUrl = 'https://www.freecodecamp.org/donate/';
+        const successUrl = 'https://www.freecodecamp.org/news/thank-you-for-donating/';
+        let stripe;
+
+        document.getElementById("donation-banner-link").addEventListener("click", onClick, false);
+
+        function onClick(e) {
+            e.preventDefault();
+            if (window.Stripe) {
+                console.log("loaded");
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
```

File: partials/banner.hbs [hunk 3]
```diff
@@ -1,5 +1,86 @@
...
+                handleStripeCheckoutRedirect();
+            } else if (document.querySelector('#stripe-js')) {
+                console.log("attached");
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
```

File: partials/banner.hbs [hunk 4]
```diff
@@ -1,5 +1,86 @@
...
+                document
+                    .querySelector('#stripe-js')
+                    .addEventListener('load', handleStripeLoadandRedirect);
+            } else {
+                console.log('not loaded');
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
```

File: partials/banner.hbs [hunk 5]
```diff
@@ -1,5 +1,86 @@
...
+            s.id = 'stripe-js';
+            s.key = 'stripe-js';
+            s.async = true;
+            s.src = 'https://js.stripe.com/v3/';
+            s.onload = onload;
+            document.getElementsByTagName('head')[0].appendChild(s);
+        };
+
+        const loadAndInitializeStripe = () => stripeScriptLoader(() => {
+            console.log('loading...');
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
```

File: partials/banner.hbs [hunk 6]
```diff
@@ -1,5 +1,86 @@
...
+    if (loadStripeList.includes(`{{@site.lang}}`)) {
+
+        // Set keys based on environement
+        const stripePublicKey = location.hostname === "localhost" || location.hostname === "127.0.0.1" ?
+            "pk_test_xM2wpbgtyotYqFCK6Ss6noxP008nxRyaE9" : "pk_live_E6Z6xPM8pEsJziHW905zpAvF";
+        const donatePageUrl = 'https://www.freecodecamp.org/donate/';
+        const successUrl = 'https://www.freecodecamp.org/news/thank-you-for-donating/';
+        let stripe;
+
+        document.getElementById("donation-banner-link").addEventListener("click", onClick, false);
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]:
```suggestion
        document.getElementById("donation-banner-link").addEventListener("click", onClick);
```
I think the options would need to be an object.

File: partials/banner.hbs [hunk 7]
```diff
@@ -1,5 +1,86 @@
...
+<script>
+    // Conditionally load stripe
+    const loadStripeList = ['en'];
+    console.log(loadStripeList.includes(`{{@site.lang}}`));
+    console.log(`{{@site.lang}}`);
+
+    if (loadStripeList.includes(`{{@site.lang}}`)) {
+
+        // Set keys based on environement
+        const stripePublicKey = location.hostname === "localhost" || location.hostname === "127.0.0.1" ?
```
Comment (ojeytonwilliams) [inline partials/banner.hbs]: Don't we want a different key for staging?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by scissorsneedfoodtoo** [comment]
One quick question: will we change the banner for all languages? If so, I could ask for translations for the banner text.

**Comment by ahmaxed** [comment]
@scissorsneedfoodtoo, no it only changes English.
Although, I should add the curriculum nav item instead of donate for english as well.

File: partials/banner.hbs (hunk 7 above)
Comment (scissorsneedfoodtoo) [reply partials/banner.hbs]: That's a good point. Seems like Stripe on freecodecamp.dev/donate is using a test key.

If the Stripe test key is the same for localhost and .dev, maybe this line could be something like:

```suggestion
        const stripePublicKey = location.hostname === "localhost" ||
            location.hostname === "127.0.0.1" || 
            location.hostname === "www.freecodecamp.dev" ? 
            "pk_test_xM2wpbgtyotYqFCK6Ss6noxP008nxRyaE9" : "pk_live_E6Z6xPM8pEsJziHW905zpAvF";
```

File: partials/banner.hbs [hunk 8]
```diff
@@ -1,5 +1,80 @@
...
+        {{{t "Help our nonprofit pay for servers. <span>Donate Now</span>."}}}
     </p>
 </a>
+<script>
+    // Conditionally load stripe
+    const loadStripeList = ['en'];
+
+    if (loadStripeList.includes(`{{@site.lang}}`)) {
+
+        // Set keys based on environement
```
Comment (scissorsneedfoodtoo) [inline partials/banner.hbs]:
```suggestion
        // Set keys based on environment
```

File: partials/banner.hbs [hunk 9]
```diff
@@ -1,5 +1,80 @@
...
+        document.getElementById("donation-banner-link").addEventListener("click", onClick);
+
+        function onClick(e) {
+            e.preventDefault();
+            if (window.Stripe) {
+                handleStripeCheckoutRedirect();
+            } else if (document.querySelector('#stripe-js')) {
+                document
+                    .querySelector('#stripe-js')
+                    .addEventListener('load', handleStripeLoadandRedirect);
```
Comment (scissorsneedfoodtoo) [inline partials/banner.hbs]:
```suggestion
                    .addEventListener('load', handleStripeLoadAndRedirect);
```

File: partials/banner.hbs [hunk 10]
```diff
@@ -1,5 +1,80 @@
...
+            handleStripeLoad();
+        });
+
+        const handleStripeLoad = () => {
+            stripe = window.Stripe(stripePublicKey);
+        }
+
+        loadAndInitializeStripe();
+
+        const handleStripeLoadandRedirect = () => {
```
Comment (scissorsneedfoodtoo) [inline partials/banner.hbs]:
```suggestion
        const handleStripeLoadAndRedirect = () => {
```

**Review by scissorsneedfoodtoo — CHANGES_REQUESTED** [review]
Sorry if this seems nit picky, but I noticed a couple of other minor things. Besides that and the test key bit, everything LGTM and worked well in testing.

File: partials/banner.hbs (hunk 7 above)
Comment (ahmaxed) [reply partials/banner.hbs]: sure, I will take care of that.

Comment (ahmaxed) [reply partials/banner.hbs]: on a second thought, maybe I should just check for .org and return the live keys.

Comment (scissorsneedfoodtoo) [reply partials/banner.hbs]: Awesome, that sound much cleaner.

File: partials/banner.hbs [hunk 11]
```diff
@@ -1,5 +1,80 @@
...
 </a>
+<script>
+    // Conditionally load stripe
+    const loadStripeList = ['en'];
+
+    if (loadStripeList.includes(`{{@site.lang}}`)) {
+
+        // Set keys based on environment
+        const stripePublicKey = location.hostname === location.hostname === "www.freecodecamp.org" ?
+            "pk_live_E6Z6xPM8pEsJziHW905zpAvF" : "pk_test_xM2wpbgtyotYqFCK6Ss6noxP008nxRyaE9";
```
Comment (scissorsneedfoodtoo) [inline partials/banner.hbs]:
```suggestion
        const stripePublicKey = location.hostname === "www.freecodecamp.org" ?
            "pk_live_E6Z6xPM8pEsJziHW905zpAvF" : "pk_test_xM2wpbgtyotYqFCK6Ss6noxP008nxRyaE9";
```

**Review by scissorsneedfoodtoo — CHANGES_REQUESTED** [review]
Noticed one more thing. Otherwise only the test key will load.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Everything LGTM :+1: :+1:

**Pull request merged**